### ISSUE TITLE
Viewpoint test checker: Missing viewpoint adaptation for superclass type arguments during subtype checking

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#1801]
I think viewpoint adaptation is missing when expanding superclass instantiations during subtype checking.

A minimal example using the viewpoint test checker:

```java
import viewpointtest.quals.*;

public class SuperclassVpaMinimal {
    static class Super<T> {}

    static class Sub extends Super<@ReceiverDependentQual Object> {}

    void test(@A Sub a) {
        // Expected: OK
        @A Super<@A Object> s = a;

        // Actual: assignment.type.incompatible
    }
}
```
Error msg:
```
/private/tmp/viewpoint-minimal/SuperclassVpaMinimal.java:9: error: [assignment.type.incompatible] incompatible types in assignment.
        @A Super<@A Object> s = a;
                                ^
  found   : @A Sub
  required: @A Super<@A Object>
1 error
```

The declared superclass type:

```java
Super<@ReceiverDependentQual Object>
```

is written from the viewpoint of the `Sub` object. When checking an expression of type `@A Sub` against a supertype, that superclass instantiation should be interpreted from the `@A` viewpoint. So the inherited supertype should be adapted to:

```java
@A Super<@A Object>
```

This is analogous to the Universe/GUT rule for superclass expansion. If a class is declared as:

```java
class D extends C<rep Data> {}
```

then when checking a value of type:

```java
peer D
```

the superclass instantiation `C<rep Data>` is not used literally. The type argument `rep Data` was written from `D`’s viewpoint, so it must be viewpoint-adapted through `peer`:

```text
peer ▷ rep Data = lost Data
```

yielding:

```text
peer D <: peer C<lost Data>
```

The same principle applies here: `@ReceiverDependentQual` in the superclass type argument is written inside the subclass declaration, so when `Sub` is viewed as `@A Sub`, the superclass type argument should be adapted from that viewpoint.
